.github/workflows/tests.yml

```diff
@@ -181,77 +184,75 @@ jobs:
 
-      - name: Setup miniconda
-        uses: conda-incubator/setup-miniconda@v2
+      - name: Setup Miniconda
+        uses: conda-incubator/setup-miniconda@v3
         with:
-          condarc-file: .\ci\github\.condarc
-          python-version: ${{ matrix.python-version }}
+          condarc-file: .github\condarc
           run-post: false  # skip post cleanup
 
-      - name: Setup environment
-        shell: cmd /C CALL {0}
-        run: |
-          @echo on
-          CALL choco install visualstudio2017-workload-vctools || exit 1
-          CALL conda install -q -y -c defaults ^
-            --file .\tests\requirements.txt ^
-            --file .\tests\requirements-windows.txt ^
-            ${{ env.CONDA_CHANNEL_LABEL }}::conda || exit 1
-          CALL pip install -e . --no-deps || exit 1
-
-      - name: Show info
-        run: |
-          conda info -a
-          conda list --show-channel-urls
+      - name: Choco Install
+        run: choco install visualstudio2017-workload-vctools
 
-      - name: Run tests
-        run: |
-          pytest `
-            --color=yes `
-            -v `
-            -n "${{ env.PYTEST_NUMPROCESSES }}" `
-            --basetemp "${{ runner.temp }}\${{ matrix.test-type}}" `
-            --cov conda_build `
-            --cov-append `
-            --cov-branch `
-            --cov-report xml `
-            --replay-record-dir="${{ env.REPLAY_DIR }}" `
-            --replay-base-name="${{ env.REPLAY_NAME }}" `
-            -m "${{ env.PYTEST_MARKER }}" `
-            .\tests
-
-      - uses: codecov/codecov-action@v3
+      - name: Conda Install
+        run: conda install
+          --yes
+          --file tests\requirements.txt
+          --file tests\requirements-${{ runner.os }}.txt
+          python=${{ matrix.python-version }}
+          ${{ env.CONDA_CHANNEL_LABEL }}::conda
+
+      # TODO: how can we remove this step?
+      - name: Install Self
+        run: pip install -e .
+
+      - name: Conda Info
+        run: conda info --verbose
+
+      - name: Conda List
+        run: conda list --show-channel-urls
+
+      - name: Run Tests
+        run: pytest
+          --cov=conda_build
+          -n "${{ env.PYTEST_NUMPROCESSES }}"
+          -m "${{ env.PYTEST_MARKER }}"
+
+      - name: Upload Coverage
+        uses: codecov/codecov-action@v4
         with:
-          flags: ${{ matrix.test-type }},${{ matrix.python-version }},win-64
+          flags: ${{ runner.os }},${{ runner.arch }},${{ matrix.python-version }},${{ matrix.test-type }}
 
-      - name: Upload Pytest Replay
+      - name: Upload Test Results
         if: '!cancelled()'
-        uses: actions/upload-artifact@v3
+        uses: actions/upload-artifact@v4
         with:
-          path: ${{ env.REPLAY_DIR }}
-          name: ${{ env.REPLAY_NAME }}-${{ matrix.test-type }}
+          name: test-results-${{ env.HASH }}
+          path: |
+            .coverage
+            tools\durations\${{ runner.os }}.json
```

Comment:
We don't have durations yet

Reply:
See https://github.com/conda/conda-build/pull/5146/commits/600d868b1d436497d7b823b3e8d9a29c71023a66